### PR TITLE
feat: enable unidirectional vnet peering

### PR DESCRIPTION
**Pull request by jaredfholgate**
# Overview/summary

This change adds a new attribute to the vnet variable to support unidirectional peering.

## This PR fixes/adds/changes/removes

N/A

### Breaking changes

None, defaults to `both` which was the previous behaviour.

## Testing evidence

Unit tests have been added to cover both cases.

## As part of this pull request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-lz-vending/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-lz-vending/issues), for tracking and closure.
- [x] Run and `make fmt` & `make docs` to format your code and update documentation.
- [x] Created unit and deployment tests and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

File: modules/virtualnetwork/locals.tf [hunk 1]
```diff
@@ -28,6 +36,8 @@ locals {
         this_resource_id   = v.hub_network_resource_id
         remote_resource_id = azapi_resource.vnet[k].id
       }
+      peering_direction   = contains(local.valid_peering_directions, coalesce(lower(v.hub_peering_direction), local.peering_direction_both)) ? coalesce(lower(v.hub_peering_direction), local.peering_direction_both) : local.peering_direction_both
```
Comment (matt-FFFFFF) [inline modules/virtualnetwork/locals.tf:39]: What are your thoughts on using a variable validation rule here, with for loop and alltrue() Could we simplify this?

File: tests/virtualnetwork/virtualnetwork_test.go [hunk 2]
```diff
@@ -335,6 +335,84 @@ func TestVirtualNetworkCreateValidWithHubPeeringCustomNames(t *testing.T) {
...
+	primaryvnet["hub_peering_direction"] = "tohub"
+	primaryvnet["hub_peering_name_tohub"] = "test-tohub"
+	primaryvnet["hub_peering_name_fromhub"] = "test-fromhub"
+
+	test, err := setuptest.Dirs(moduleDir, "").WithVars(v).InitPlanShowWithPrepFunc(t, utils.AzureRmAndRequiredProviders)
+	require.NoError(t, err)
+	defer test.Cleanup()
+
+	// We want 10 resources here, 2 more than the TestVirtualNetworkCreateValid test
+	// The additional two are the inbound & outbound peering
```
Comment (matt-FFFFFF) [inline tests/virtualnetwork/virtualnetwork_test.go:358]:
```suggestion
	// We want 9 resources here, 1 more than the TestVirtualNetworkCreateValid test
	// The additional one is the outbound peering
```

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

File: modules/virtualnetwork/locals.tf (hunk 1 above)
Comment (matt-FFFFFF) [reply modules/virtualnetwork/locals.tf:39]: Ok fair, happy with this

**Comment by matt-FFFFFF** [comment]
Doing specific deployment test for VirtualNetwork https://github.com/Azure/terraform-azurerm-lz-vending/actions/runs/6096771492

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

File: tests/virtualnetwork/virtualnetworkDeploy_test.go [hunk 3]
```diff
@@ -129,6 +129,47 @@ func TestDeployVirtualNetworkValidVnetPeering(t *testing.T) {
 	test.ApplyIdempotent().ErrorIsNil(t)
 }
 
+// TestDeployVirtualNetworkValidUniDirectionalVnetPeering tests the deployment of a virtual network
+// with unidirectional peering to a hub virtual network.
+func TestDeployVirtualNetworkValidUniDirectionalVnetPeering(t *testing.T) {
+	t.Parallel()
+
+	utils.PreCheckDeployTests(t)
+	testDir := "testdata/" + t.Name()
```
Comment (matt-FFFFFF) [inline tests/virtualnetwork/virtualnetworkDeploy_test.go:138]:
```suggestion
	testDir := ""
```

Comment (matt-FFFFFF) [reply tests/virtualnetwork/virtualnetworkDeploy_test.go:138]: @jaredfholgate please can you fix this to enable the deployment tests to run?

The test dir is inly required when calling the module from another root module.

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

**Comment by jaredfholgate** [comment]
@matt-FFFFFF Apologies for the delay. I have made the fix to the test and taken latest from main.

**Pull request merged**